docs/best-practice-consensus-pipeline.md

```diff
@@ -0,0 +1,217 @@
+<img src="https://github.com/fulcrumgenomics/fgbio/blob/main/.github/logos/fulcrumgenomics.svg" alt="Fulcrum Genomics" height="35"/>
+
+# fgbio Best Practise FASTQ->Consensus Pipeline
+
+This document describes the best practise(s) for building pipelines to go from FASTQ files that contain a mixture of template (genomic) bases and UMI bases, through to aligned and filtered consensus reads.  
+
+The pipeline is broken into two phases:
+
+1. Getting to aligned and grouped raw reads
+2. Going from raw reads to aligned and filtered consensus reads
+
+We provide two alternatives for the second phase.  One intended for R&D environment where it is desirable to be able to branch off from the pipeline and test e.g. multiple consensus calling or filtering parameters.  The other is intended for high throughput production environments where performance and throughput take precedence over flexibility.
+
+## Tools used
+
+The steps described below use a mixture of:
+
+- `fgbio` (version 2.0 or higher)
+- `samtools` (version 
+- `bwa mem`
+
+
+## Phase 1: FASTQ -> Grouped BAM
+
+The first phase deals with going from FASTQ files through to aligned and grouped BAM files.  The general structure of this phase is:
+
+```mermaid
+graph TD;
+A["fgbio FastqToBam"]-->B["samtools fastq | bwa mem | fgbio ZipperBam"];
+B-->C["fgbio GroupReadsByUmi"];
+```
+
+In this example we'll use a single sample with paired-end fastq files and an inline 8bp UMI at the beginning of read one (see the fgbio [Read Structure](https://github.com/fulcrumgenomics/fgbio/wiki/Read-Structures) page for how to specify other UMI setups).
+
+### Step 1.1: FASTQ -> uBam
+
```

Comment:
This step extracts the UMI bases and stores them in standard SAM tags.  
This is preferred over non-standard solutions that store UMIs in FASTQ read names or otherwise.